openlibrary/plugins/worksearch/code.py

```diff
@@ -1046,7 +1049,8 @@ def work_search(query, sort=None, page=1, offset=0, limit=100, fields='*', facet
         sort = process_sort(sort)
 
     # deal with special /lists/ key queries
-    query['q'], page, offset, limit = rewrite_list_editions_query(
+    query['q'], page, offset, limit = cache.memcache_memoize(
+        "rewrite_list_editions_query", "search.list_query", timeout=5*60)(
```

Comment:
```suggestion
        rewrite_list_editions_query, "search.list_query", timeout=5*60)(
```
I think that the first parameter should be a callback, not a string.

Should `"search.list_query"` be replaced with something that uniquely identifies the query?